### PR TITLE
Made adb present check part of instantiation

### PR DESCRIPTION
**Pull request by 0x1mason**
Created a factory method to consolidate instantiation with checking adb exists.

@jlipps @sebv The taking of ganders, please if you will.

### CODE REVIEW

File: lib/adb.js [hunk 1]
```diff
@@ -46,10 +47,16 @@ var ADB = function (opts) {
   if (opts.remoteAdbPort) {
     this.adb.defaultArgs.push("-P", opts.remoteAdbPort);
   }
+  var initError = null;
+  async.series([
+    this._checkAdbPresent.bind(this)
+  ], function(err) {
+    initError = err;
+  });
+  if (initError) throw initError;
```
Comment (jlipps) [inline lib/adb.js]: there's an async bug here: this line will usually execute before `this._checkAdbPresent` ever has a chance to.

we can't run async functions inside a constructor. if we want to do something like this we should create an `init` function or something which is async

Comment (0x1mason) [inline lib/adb.js]: Ok, I know what I'll do.

Comment (sebv) [inline lib/adb.js]: Was also wondering about this `async.series`.

**Comment by 0x1mason** [comment]
@jlipps @sebv Let's try again.

**Pull request merged**